archive_upload/handlers/dsmc_handlers.py

```diff
@@ -468,6 +469,12 @@ def post(self, runfolder_archive):
         log.info("Uploading {} to PDC...".format(path_to_archive))
         cmd = "export DSM_LOG={} && dsmc archive {}/ -subdir=yes -description={}".format(
             dsmc_log_dir, path_to_archive, uniq_id)
+
+        # Mock starting the TSM process if mock mode is enabled
+        tsm_mock_enabled = self.config.__getitem__("tsm_mock_enabled")
```

Comment:
I'm pretty sure that `__getitem__` shouldn't be called directly, it's a hidden implementation detail and I think it's possible that you can have cases where this wouldn't return the expected value. Instead I would use `self.config["key"]` or `self.get("key", some_default)`.

Reply:
... in this case, I think you want the latter one, i.e.: `self.get("tsm_mock_enabled", None)`.

Reply:
This no worky since config is not a dict, but an instance of a custom wrapper class. As we discussed, I'm implementing a try-except solution for the optional config key problem.